R/uniDimBayesianCalcCoeff.R

```diff
@@ -569,3 +543,107 @@
+    return(.getStateContainerB(jaspResults)[["scaleResultsObj"]]$object)
+
+  out <- model[["scaleResults"]]
+  if (is.null(out))
+    out <- list()
+
+  if (is.null(model[["empty"]])) {
+
+    ciValue <- options[["credibleIntervalValueScale"]]
+
+    selected <- names(which(model[["derivedOptions"]][["selectedEstimators"]]))
+    # first the coefficients with samples
+    samps <- model[selected]
+    samps <- lapply(samps, function(x) x[["samp"]])
+    samps[sapply(samps, is.null)] <- NULL
+
+    out[["est"]] <- lapply(samps, mean)
+    out[["cred"]] <- lapply(samps, function(x) coda::HPDinterval(coda::mcmc(c(x)), prob = ciValue))
+
+    if (options[["rHat"]]) {
+      tmp <- lapply(samps, function(x) {lapply(as.data.frame(t(x)), coda::mcmc)})
+      out[["rHat"]] <- lapply(tmp, function(x) {coda::gelman.diag(coda::as.mcmc.list(x))[["psrf"]][, 1]})
+    }
+
+    # check for mean and sd
+    if ("meanScale" %in% selected) {
+      out[["est"]][["meanScale"]] <- model[["meanScale"]][["est"]]
+      out[["cred"]][["meanScale"]] <- model[["meanScale"]][["cred"]]
+      if (options[["rHat"]]) {
+        out[["rHat"]][["meanScale"]] <- NA_real_
+      }
+    }
+    if ("sdScale" %in% selected) {
+      out[["est"]][["sdScale"]] <- model[["sdScale"]][["est"]]
+      out[["cred"]][["sdScale"]] <- model[["sdScale"]][["cred"]]
+      if (options[["rHat"]]) {
+        out[["rHat"]][["sdScale"]] <- NA_real_
+      }
+    }
+
+    stateContainer <- .getStateContainerB(jaspResults)
+    stateContainer[["scaleResultsObj"]] <- createJaspState(out, dependencies = c("credibleIntervalValueScale",
+                                                                                 "meanScale", "sdScale", "rHat",
+                                                                                 "alphaScale", "omegaScale",
+                                                                                 "lambda2Scale", "lambda6Scale",
+                                                                                 "glbScale","averageInterItemCor",
+                                                                                 "meanMethod", "sdMethod"))
+
+  }
+
+  return(out)
+
+}
+
+
+.BayesianComputeItemResults <- function(jaspResults, options, model) {
+  if (!is.null(.getStateContainerB(jaspResults)[["itemResultsObj"]]$object))
+    return(.getStateContainerB(jaspResults)[["itemResultsObj"]]$object)
+
+  out <- model[["itemResults"]]
+  if (is.null(out))
+    out <- list()
+
+  chosen <- any(model[["derivedOptions"]][["itemDroppedSelected"]])
+
+  if (is.null(model[["empty"]]) && chosen) {
+
+    ciValue <- options[["credibleIntervalValueItem"]]
+
+    selected <- names(which(model[["derivedOptions"]][["itemDroppedSelected"]]))
+    # first the coefficients with samples
+    samps <- model[selected]
+    samps <- lapply(samps, function(x) x[["itemSamp"]])
+    samps[sapply(samps, is.null)] <- NULL
```

Comment:
see the previous comment. This could also be a function since you do this twice.